### ISSUE TITLE
Citation of OpenRSP program/library

### ISSUE THREAD

**Issue by bingao** [openrsp/openrsp#43]
Hi @bast and @ringholm ,

I think we may need a more appropriate way to cite OpenRSP program/library (at least, a DOI needed).

Now in `doc/citations.rst`, I just simply copied and pasted from LSDalton manual regarding OpenRSP:

> OpenRSP, an open-ended response property library, www.openrsp.org

Therefore, could you please fix it some time?

**Comment by bast** [comment]
I will get the DOI once we release but we need a collective decision about how we wish to be cited.

**Comment by bingao** [comment]
Thanks. I clear the assignees so that a collective decision is required as you suggested.

**Comment by foeroyingur** [comment]
Sorry for going a bit off topic. I'm wondering if citation file format ([CFF](https://citation-file-format.github.io/)) is relevant, @bast? I was considering using it for other projects but not sure whether it has been adopted by anyone?

**Comment by bast** [comment]
Yes we should use it, I discussed with Stephan two days ago :-) But I think we should also recommend a citation that people can put in their papers. I wanted to work from this side but also add CFF and DOI.

**Comment by bast** [comment]
After some more thinking I think CFF is the way to go.

**Comment by ringholm** [comment]
The present version of the citation guide is found at http://www.openrsp.org/en/latest/citations.html and should be updated once the present issue is dealt with.

**Comment by ringholm** [comment]
@bast Has everything that you need for this issue been decided?

**Comment by bast** [comment]
Is this list complete: http://www.openrsp.org/en/latest/authors.html?
If yes, then yes :-)

**Comment by ringholm** [comment]
No, I will look at it now. It's better to have too many than too few authors on this list - but just as a guideline, do you think it's reasonable to include on this list authors who could have made some contribution a while back but who don't have any active code on master? I am thinking about e.g. Orian Louant who made some contributions in the legacy version but who is not active now, nor is I think his code used in the most updated version.

**Comment by bingao** [comment]
IMO, we should have a list of previous authors if we want to add someone who does not have active codes any more and who is not involved into OpenRSP development any more.

**Comment by ringholm** [comment]
This appears to be the list I used for the license proposal:

```
Name					Approved?
Andreas J. Thorvaldsen			Yes
Arnfinn					Yes
Bin Gao					Yes
Dan Jonsson				Yes
Daniel Henrik Friese			Yes
Jógvan Magnus Haugaard Olsen		Yes
Kenneth Ruud				Yes
Orian Louant				Yes
Radovan Bast				Yes
Simen Sommerfelt Reine			Yes
Thomas Kjaergaard			Yes
```

Do we know about any people not on this list who should be mentioned in the authors list? After thinking about this just now, I think that maybe Yann Cornaton should be listed here and I have written to him and asked him about what he contributed. I think he could have contributed to some transformation routines that are not in use in the new version, but if so, I don't understand why he would not show up on the grep of authors that we used as a basis for the above list.

**Comment by ringholm** [comment]
Also there is the matter of Thomas Kjærgaard whose contribution to OpenRSP is unknown, although I think that he was listed in the grep of authors. I have pushed a new list of authors to the branch `website-authors-update` that errs on the side of caution.

**Comment by ringholm** [comment]
I don't know the present affiliation of @tkjaergaardchem  and @arnfinn: Thomas, could you let us know if you should be listed as an author of OpenRSP and if so, your current affiliation, and Arnfinn, should I put UiT as your current affiliation? (Do you still have a part-time position there?)

**Comment by bast** [comment]
For the citation part the affiliation is less urgent. Most important is that we get the list of names. Also I assume alphabetical order? (Any order is fine for me)

**Comment by ringholm** [comment]
I have used alphabetical order of last name. We will PR the updated author list today or tomorrow.

**Comment by ringholm** [comment]
I have added three more authors: Marco Anelli, Karen Dundas and Nanna H. List. Not clear if all of these should be considered as authors but I am erring on the side of caution: Marco said that he has implemented some properties (I am asking him to tell a little more about it), Karen has not as far as I know contributed to the OpenRSP repository but has done quite a lot of work with callbacks to FraME for OpenRSP usage in LSDalton, and I think that Nanna has done some work on earlier implementations with PE contributions (I have also e-mailed her to ask about it).

**Comment by bast** [comment]
Thanks! From my perspective: yes.

**Comment by ringholm** [comment]
@bast A complication concerning the new authors that now emerged is that not all of them were actually asked about the licensing. First of all, do you agree with my sentiment that these should now be considered authors? Secondly, do you think as a consequence that we should ask for their agreement about releasing under LGPLv2.1 before we mint the release?

**Comment by arnfinn** [comment]
@ringholm: I do not have a position at UiT anymore, but please use UiT as my affiliation.

**Comment by bingao** [comment]
May I ask how we define an author of OpenRSP? Personally, I like to distinguish previous authors and current authors.

**Comment by ringholm** [comment]
Update: Marco gave his approval in a merge request discussion earlier. I have asked Karen and Nanna for their approval of the licensing proposal that was sent out in the first round of asking for such approvals. We did already do some of what we asked about in their proposal but I think the best we can do now is to ask for their approval about all of it.

@bast Do you agree that we should wait until they reply? Unfortunate that this happens so late but maybe they will respond quickly.

**Comment by ringholm** [comment]
@bingao Right now I have used what I think is a permissive definition about an author: It could involve both

* Writing routines that are part of the present `master` branch or any earlier version of code that was written "inside" the OpenRSP project (not necessarily just the OpenRSP repository)
* Done supervision of the project or contributed with advice or discussion to a reasonable degree

I am not sure if I can easily make a clear distinction between current and previous authors. I agree that this is something we can look at but I am not sure if it is crucial for the release.

**Comment by bingao** [comment]
Hei @ringholm Thank you for the clarification.

As regarding the first point, I personally do not understand the reason of "not necessarily just the OpenRSP repository", but to the OpenRSP project. Because I understand what we are talking about is the authors of OpenRSP library, as such I feel it is natural to restrict ourselves to OpenRSP repository.

Just my personal opinion.

**Comment by bast** [comment]
OK it seems there are 4 questions:
1) who should be authors
2) do we need more fine-grain (previous/present, author/contributor)
3) do all authors agree to release the code under LGPL
4) are all authors informed that they are actually authors and do all authors agree to be authors

I consider 3) to be a done fact and would not like to re-discuss it and I hope that the remaining persons simply agree with the choices that have been made.

As to 4) it would be great if the persons reply before Sunday morning so that I am not put in a difficult situation.

For 1) and 2): I don't know, no strong opinion from my side and my hope is that we can perhaps postpone any changes for after `1.0.0-alpha` and fix it then for `1.0.0`.

Tomorrow is another day, hopefully it has answers for us.

**Comment by ringholm** [comment]
OK, so for 1) I would say that the definition of being an author was opened a bit after the meeting on Monday where the consensus seemed to me to be that Simen should also be counted among the authors. He has not to my recollection written any of the code in the OpenRSP repository. If we use your definition @bingao, then he should not be counted among the authors. But if we use a more open definition and Simen is included, then that in turn would imply that it is a fair claim that Karen should also be counted an author. Basically, what I thought was that a) it was better to be permissive with the definition of authorship rather than too strict, and b) it doesn't really detract from the authorship of the people who have contributed most that more people are counted as authors.

For 2) I am fine with looking into a categorization of authors but I do not really want to be the person who makes it. I also think that this is not crucial for the `1.0.0-alpha` release.

For 3) it looks like all the known authors except Karen have made statements to that effect. She has been contacted and I hope she replies during the course of the day.

For 4), I cannot answer that question with 100% certainty as I don't have an exhaustive overview of the contributors. The previous round of approval-seeking for the license proposals was done based on a grep of authors that you @bast made and considered to be safe to base ourselves on (see https://gitlab.com/openrsp/lsdalton/merge_requests/32 ). In retrospect it appears that there may have been a couple more people as well. I cannot at the moment recall anyone else who might have contributed, except for possibly Michal Jaszunski during the work on pure vibrational (hyper)polarizabilities. I don't remember exactly if he was somehow involved in some code to process the response tensors into observables and I am not sure if we can find this out on such short notice. It's very difficult for me to conclude at any time that there are definitely no more persons to be considered as authors since there are always the possibility that we forgot someone or that someone contributed something before I became involved in the project. I will happily take any advice about what to do about this.

**Comment by bast** [comment]
If I am to create a citation recommendation tomorrow, I think it is simply too late to now discuss who should be authors and who should not. This should have been started weeks ago.

With this I see the following options:
1) We list http://www.openrsp.org/en/latest/authors.html and add more authors for 1.0.0.
2) We list everybody who has been mentioned in this thread and adjust for the 1.0.0 release.
3) I create the DOI but do not list the authors explicitly and we fix that for the 1.0.0 release.
4) I do not create any OpenRSP release and on the LSDalton side we reference a commit and "hope that nobody tries to cite OpenRSP before 1.0.0" (I know this sounds absurd).
5) We postpone.

I know that even deciding among these options is difficult.

**Comment by ringholm** [comment]
I don't know what to do and I don't feel competent to make this decision by myself.

**Comment by bingao** [comment]
My suggestion is that we should not do anything if we do not know. I suggest that we need to vote the decision afterwards, the same as we did for the license.

**Comment by bast** [comment]
Since we have doubt and little time, my recommendation is that I create tomorrow the tag, get the DOI - this is the right thing from the reproducibility perspective - but I keep the authors undefined/vague until we settle this for the next tag. With this I will also postpone #69.

And "not do anything" means really nothing? No release tag?

**Comment by bingao** [comment]
Again, I am strict on the definition of authors of a library (please note I am not talking the project): we are release it, we create DOI, it looks like we publish an article. Therefore, I vote for a strict definition of OpenRSP library authors.

**Comment by ringholm** [comment]
Concerning the discussion about who should be authors - assuming that there is a 1:1 relationship about agreement to the license proposal and the authorship, I did consider that question settled weeks ago after taking your (@bast 's) advice about sending the license proposal to the people on the author grep list. The point was opened again after the question was put to the last OpenRSP meeting about whether or not the list of authors on the OpenRSP website was updated. It was then brought up that Simen should maybe be on that list. The consensus seemed to me to be that he should indeed be. I guess that's why I started to rethink this at this late stage and that's why it was not done before.

**Comment by bast** [comment]
For me DOI means: make this thing findable 15 years from now. It does not have to mean that we published an official release.

**Comment by bingao** [comment]
Sorry. Correct myself. I mean do nothing as regarding the names of authors.

**Comment by ringholm** [comment]
I can also add that I have already contacted Karen, Nanna and Marco and told them that we intend to make them authors. Nanna has already replied and also agreed to the licensing proposal.

**Comment by bast** [comment]
As to "nothing as regarding the names of authors" - what does this mean for the relase tag:
- no release tag?
- release tag but no DOI?
- DOI but no explicit names?

**Comment by ringholm** [comment]
I have a branch `website-authors-update` which contains an authors.rst with more authors than what is presently on the website. It can potentially be merged in but I cannot say if it is an exhaustive list or not.

**Comment by bast** [comment]
I have a feeling that a wide and very exhaustive flat author list is not the ideal solution for all authors (it is totally fine for me), since some have clearly contributed "a lot more" to the library than others. I feel there is an unspoken issue here that we will not solve within 24 hours.

What can we do in practical release terms if we accept that we haven't settled the list of authors?

**Comment by ringholm** [comment]
@bast I agree with the sentiment that we will likely not solve this within 24 hrs. I am also not sure if we will ever be able to definitely conclude that everyone who ever concluded anything is accounted for.

Accepting that we haven't settled the list of authors, a problem is that we don't know whose approval we should have sought before making the code public, and this problem is brought to a head when there is the matter of making an official release, be it alpha or not. We may have already made public code without the approval of some who should possibly have had a say in it, and from that perspective it would be a further step in the wrong direction to make an official release without taking care of this.

**Comment by ringholm** [comment]
If I were someone whose authorship status in this is unsettled and if I was contacted now with this, I would in all likelihood say "go ahead and do what you like" but I cannot just assume this for others.

**Comment by bast** [comment]
OK but at least for me neither a release tag, nor DOI have to mean official release. For me these are technical tools which make it easier to find a specific version many years from now.

We can postpone the authors part, does that sound reasonable or should we postpone the tag and the DOI as well?

**Comment by ringholm** [comment]
But making OpenRSP part of an LSDalton release is official, is it not?

**Comment by bast** [comment]
By making OpenRSP part of LSDalton release we make it easier for others to use OpenRSP. Whether LSDalton references 5d1e4b1bc7d3d3707f69669902f7114f207f7abf or `1.0.0-alpha` makes little difference to the code and also little difference to the user of LSDalton.

The important thing for LSDalton users is to make it easy for them to cite OpenRSP work.
The important thing for OpenRSP authors is to get credit for their work.

Whether the release is official or not is more important for other codes because they will want to interface to something that is expected to be stable and not change API the next day. Not an issue for LSDalton right now since Simen knows.

**Comment by bast** [comment]
The one real-life danger is that somebody downloads LSDalton 2018.0, uses it, never upgrades, never checks the websites, the output says nothing because we were not sure, hence the user will not cite OpenRSP.

**Comment by ringholm** [comment]
OK - I need to go in a short while and I will then likely not be available until the evening. I don't know what to do here and I don't feel as if I know enough about this to make a properly informed decision, and I am also afraid of making the wrong decision - basically I feel trapped. I still have the feeling that no matter what we call the making public of OpenRSP with LSDalton, it is a release, and for those possible authors who have not approved, it would be wrong to them if we take that step. To add to this, I don't know if we ever will be able to name everyone who might have made some contribution.

As for the real-life danger, we currently call what we plan to do a "limited release" - see https://gitlab.com/dalton/lsdalton/merge_requests/49 - but we also make a strong discouragement about using that code for production calculation.

**Comment by bast** [comment]
My recommendation:
- we create the tag
- we create the DOI but without explicit authors
- in the LSDalton output we refer to http://www.openrsp.org/en/latest/citations.html which allows us to tweak it later (we probably already do)

I am not too worried about the list being exhaustive enough: I think that those who are invested in this project, have been at least contacted or have spoken.

I am more worried that the list is too flat and possibly lowers credit for e.g. Gao who spent a huge effort rewriting and modularizing it.

Conclusion is I think that the authors list will not be known this weekend and we have to decide what we do given this boundary condition.

**Comment by bingao** [comment]
I am sorry that I probably misunderstood when @bast prepared the list of authors for asking for license. Does it mean:

1. Once as an author, always be an author even there is no code/document etc. left?

If yes, I would definitely suggest the distinguishing between previous and current authors.

2. To make a release, which contains no contribution from a “previous author”, but this “previous author” still have the right and obligation to vote, for example, the choice of license, and say “no” if he or she wants?

That makes me feel a bit unfair, sorry. Also, I am confused here: we release some version under some license, instead of the whole history of OpenRSP, right? If yes, why do we need to ask “previous authors” permission since their contributions will not be released in such a way?

**Comment by bast** [comment]
These are important questions which are difficult to answer. Dalton and DIRAC has been through this and it is not easy. Impossible to solve this weekend so let's not try to.

As to the present/past: it does matter what past authors have to say. If somebody new joins the team and rewrites the whole library to Python and removes all other authors except their name, we would find it perhaps unfair, too.

Credit is important and for me Gao's and Magnus' opinions here count a lot since you have been clearly invested above average. I am fine with flat but it is easy for me to say.

**Comment by ringholm** [comment]
OK, so let me summarize:

* I think that the commit we intend to ship with LSDalton does not contain any legacy code - am I correct in this? If I am correct, then this in all likelihood means that shipping this commit does not constitute any additional "damage" to any of the unsettled authors.
* Granted that the previous point is OK, then @bast I think that I am fine with your recommendation on https://github.com/openrsp/openrsp/issues/43#issuecomment-439609548 and if you do this, can you also update the citation guide in the OpenRSP part of the LSDalton manual to point to the URL as well instead of listing the articles?
* @bingao I understand your opinion about authorship and I agree with Radovan that it will be difficult to decide this now. I think that we should discuss this in more detail at the next meeting and try to settle on something that we are all happy with.

I will be back this evening for any new questions.

**Comment by foeroyingur** [comment]
Sorry, I didn't notice this discussion earlier. Personally, I would not add myself and others who have contributed only very few lines of code in an old legacy version of the code. Actually I wouldn't even do it if it wasn't legacy code. Normally, as far as I know, and it is also my opinion that, coauthorship requires some substantial contribution. Of course substantial is a matter of interpretation, but the few lines that the PE developers added inside OpenRSP at one point does not lead to coauthorship. What I perhaps would do is distinguish between authors and other contributions.

**Comment by ajthorva** [comment]
I'd say the "LGPL" and "Release author" are separate issues:

1. Do you agree that your parts of the code+docs is released under LGPL 2.1 - this only concerns people with commits in the GitHub repo (including the history that was brought over from GitLab). If someone had declined, his/her commits would have had to be removed from the history ported to GitHub.

2. Do you consent to be co-author of the 2018 (1.0.0) release of OpenRSP - this should be asked to people with lines to their name on master at the time of the release, and arguably as well those who have contributed with testing, funding or other time-consuming work for OpenRSP (people are free to decline, of course, as I probably will for the next release).

Question 1 is a legal issue, and was settled with the move to GitHub and going public.

Question 2 is an academic issue, and is settled apart from 1 or 2 names. In my opinion it's enough to have *made* the offer of authorship. If some don't respond, I would take that as a "no". I don't think one should consider work very far back in time - authors should all have known that they were working on the OpenRSP library. A couple of phone calls or SMS might settle it before Monday.

IMO separating "Authors of this release" from "Current and previous contributors" might be a good idea (but for a future release). This would make it more attractive to start contributing - you'd become 1 name in ~10 instead of 1 in ~50.

**Comment by arnfinn** [comment]
Just to add yet another post to this very long discussion:) My contributions to the code have been minimal, as far as I remember, so I am fine with not being an author of OpenRSP. Thus, I agree with @foeroyingur [comment](https://github.com/openrsp/openrsp/issues/43#issuecomment-439619364).

**Comment by bast** [comment]
Thanks for the very good input. I am still unsure what to do tomorrow with the tagging.

**Comment by foeroyingur** [comment]
The tagging itself doesn't include any author information does it?

**Comment by bast** [comment]
Tagging does not include author info - so I summarize:
- I will tag tomorrow to `v1.0.0-alpha`
- I will create a DOI
- I will send a MR to LSDalton referencing the latest commit in OpenRSP

If you disagree, please let me know.

**Comment by bingao** [comment]
Hei,

Thank you all for the valuable discussion. I agree that we can not solve this problem during weekend. I think it becomes an important issue, and we need to vote the decision afterwards, instead of few persons involved and made the decision here.

This afternoon I tried to recall my memory as regarding the list of authors prepared by Radovan some time ago. I apologize that I probably misunderstood the purpose of the list and failed to speak out ago. I understood the reason that Radovan prepared the list is because we need to move OpenRSP from Gitlab to Github, and I overlooked the license issue.

I now try to summarize my opinion, in case that I forget when we discuss in some meeting later :-)

1. I suggest that we make a clear definition as regarding who can be an author of OpenRSP program, and what is the right and obligation of an author.

2. I agree with @ajthorva and I suggest that, if allowed, we should choose authors and license per release. The choice of license is **non-retroactive**. I think this suggestion will make our life easier. We do not need to ask the permission from authors who do not have contribution to the current release.

As regarding the "contribution", it is related to my first point. Once we recoganize the rules of being an author, we will understand clearly what the contribution means.

Let me emphasize my second point again: Let us imagine, if we want to make a release of OpenRSP understand another license in the future. I think it is painful for active authors at that time to contact all previous authors to ask for permission.

Last, I would like to answer Radovan's point: "If somebody new joins the team and rewrites the whole library to Python and removes all other authors except their name, we would find it perhaps unfair, too."

Let me express in such a way, if some day, some one rewrites OpenRSP using Python and I do not contribute to the newly written Python OpenRSP. I definitely will not be the author of this version of OpenRSP. I feel it is fair to me.

**Comment by bast** [comment]
Future license change may not be possible without contacting past authors. It is complicated. But these are two different things, I agree.

Roberto reminded me of a good point: software authorship is also maintainership. Maintaining is as important as authoring. This should also play a role when deciding who is author and who is not.

**Comment by bingao** [comment]
> Future license change may not be possible without contacting past authors. It is complicated. But these are two different things, I agree.

I would like to hear the reason of this point when we have time.

Because I thought the license change is for that release, not for previous releases, and not for the repository of OpenRSP. I also thought the same as the first answer of this question https://opensource.stackexchange.com/questions/33/how-can-a-project-be-relicensed

> If others contributed to your project, you have to respect their rights. So you can only assume the rights given to you by the open source license. Your options are:
>
> - contact all contributors and ask for their permission
> - switch to a compatible license (this means, if your original license is permissive, it might have a lot compatible options, if it is copyleft, the compatible options may be limited)
> - remove contributions of others before switching

If the above points hold, then I understand that one does not need to ask the permission of relicense from an author who does not have lines of codes/comments/documentation in that release. Am I right?

**Comment by ringholm** [comment]
OK, so several new comments here. I ask that we for now only concentrate on what is necessary to do before the release of LSDalton to the degree that this can be disentangled from other matters.

Concerning this matter, I am still not sure what to do before the release - I am trying here to address the new points about this:

@ajthorva, you add some useful comments. Concerning your comment 1, I think the history that was brought over from GitLab also contains code from when OpenRSP was in Dalton. I am not sure if the history of that development is available to us and if so, if it can be easily discerned OpenRSP work from non-OpenRSP work there. As you know, we gathered a collection of approvals before moving to GitHub, but that collection was based on a simple grep of commit authors in the Gitlab repository. It is now recently that I grew concerned over the matter that there could be other authors that should have been contacted.

BUT: I just remembered now that the open-source approval was gathered for Dalton by Radovan and I wonder if that would not cover the use of any such code for OpenRSP. So my first question is: 

1. @bast or @ajthorva do you have any comments on this? Was the scope of the open-source approval also inclusive of whatever work was done in OpenRSP at the time? Maybe we don't need to go through the early history.

Then, for @ajthorva's point 2:

2. Can we say anything more specific about who should be contacted about the consent to be author (I get "lines to their name on master" but I am not sure who outside this would fit into the "arguably" categories, and there might possibly also be lines on master whose origin is uncertain (I guess they could have been copied into the gitlab repo and the gitlab author of those would I assume be whoever did the copying - not necessarily the same person who wrote the lines)). And if we can make a clear list here, do we have time to get a response from everyone before Monday?

3. If we do not add any author info like @bast's summary of intentions, do we need to anything about this until Monday?

Concerning the comments made about what should constitute authorship or not: To the extent that this can be disconnected from what we need to do to release, I think we should deal with this issue at a later meeting.

Finally, concerning @bingao's point "Because I thought the license change is for that release, not for previous releases, and not for the repository of OpenRSP.". I am not sure if I understood this point correctly, but I would just add that the text of the proposal that we agreed to read "1. All applicable material associated with the OpenRSP project will be made public and get the GNU Lesser General Public License (version 2.1) software license", and "2. The OpenRSP software repository containing source code, documentation and various other associated files will move to and use Github as a hosting platform - the current platform is Gitlab. This change may take place at a different time than the adoption of point 1 (the license)." - i.e. nothing in the text was connected to a specific release although releasing was of course a motivating factor for sending out that proposal.

Finally, I should add that I am aware that I have a tendency to overthink matters like we are discussing here. If any of you feel like we are splitting hairs here and there is an obvious path forward, please let me know.

**Comment by bingao** [comment]
Hei @ringholm 

When I talked the license change, I mean the future change if it happens, not the current release. Therefore, I asked Radovan the reason of

> Future license change may not be possible without contacting past authors. It is complicated. But these are two different things, I agree.

As regarding the text of the proposal, I understand point 1 and 2 are two different things and they are not related to each other.

It might be a problem of the proposal, as I reread it now, is the point 1:

- What is "all applicable material associated with the OpenRSP project"? What is "OpenRSP project"?

Previously I understood they are exactly the "OpenRSP software repository" of the point 2.

**Comment by ringholm** [comment]
@bingao I have not defined those terms exactly. I think I intended for "all applicable material" to mean "at least the OpenRSP software repository but possibly also other material that may be reasonably associated with OpenRSP if such material exists". I am not sure if I can break the "OpenRSP project" into a more detailed definition.

**Comment by ringholm** [comment]
It's evening now and I will not stay around for much longer. I don't think I can spend that much time on this tomorrow but I will at least check in early. We should not go ahead until we have settled this.

**Comment by bast** [comment]
Let us please not discuss re-licensing now. This is a complicated matter and is not urgent. We will discuss it later - in short LGPL affects what you can do with derivative code and even when you completely rewrite it and I contribute nothing to the rewriting, you can not simply change the license to a more closed one without consulting me ("me" as placeholder for a current author, not me as Radovan).

"We should not go ahead until we have settled this." - does that mean that I should **not** create a release tag, and not create a DOI? I am again unsure about this after the recent posts here.

**Comment by ringholm** [comment]
@bast Sorry, I should have been more clear about it - maybe I lost the overview here, but I had the notion that there are still unsettled questions before you can mint the release, create the tag and create a DOI, i.e. before we can go ahead with the planned release of OpenRSP in LSDalton. See e.g. my comments in the long post above. But maybe it's best to look at this early tomorrow when at least my head might be clearer.

**Comment by bast** [comment]
OK that is totally fine. To be clear: I am on standby and **not** creating a release tag, and **not** setting any DOI until further notice.

**Comment by ringholm** [comment]
OK, let's continue tomorrow. Maybe I can ask until then whether you think any of the discussion so far or my comments in https://github.com/openrsp/openrsp/issues/43#issuecomment-439642797 should stop our release plans tomorrow.

**Comment by bast** [comment]
The move to LGPL requires approval from all authors who have contributed to the current state of the code and also from those who have contributed non-trivial patches to previous versions of the code from which the current code derived from, in other words from the copyright holders. The question who copyright holders were can be very difficult to answer.

My understanding was that we had the approval from all such authors.

If some code portions derive from the Dalton code, then we can now redistribute them since the license is compatible (actually the same). However we may have to specify the origin and cannot change the copyright for those files. Since this has never been formalized, it can again be very complicated.

**Comment by bingao** [comment]
Hei @bast 

> in short LGPL affects what you can do with derivative code and even when you completely rewrite it and I contribute nothing to the rewriting, you can not simply change the license to a more closed one without consulting me ("me" as placeholder for a current author, not me as Radovan).

Yes, I understand and I think we can discuss it later. But as you wrote here, it affects the **derivative code**. If someone redesigns the code and it becomes totally different from OpenRSP, then it will not be covered by the LGPL scope.

To make it clear: Andreas previously wrote the core part in a manual way, Magnus rewrote the core part but using recursive scheme. These two versions are different, and I do not think Magnus' codes are the derivative work on Andreas' codes, from the **technical perspective**.

Another slightly related question: you mentioned the change to a more closed license. How about the change to a more permissive one? We can discuss later.

**Comment by bast** [comment]
Has Magnus used the original code to arrive at the current code? And I mean the source code, not a specification of the code (algorithm). If yes, then this is in my understanding derivative work.

**Comment by bingao** [comment]
How about if not? Then it is not a derivative work? Or it is complicated that we would better ask a layer? (well, I do not mean we really need to ask a layer, but rather to emphasize its complexity).

**Comment by ringholm** [comment]
@bast concerning https://github.com/openrsp/openrsp/issues/43#issuecomment-439649147 I don't know the distinction between a trivial and non-trivial patch. I agree that the question of who copyright holders were could be difficult to answer. Until this discussion I had the belief that we had the approval from all the necessary authors. That was based on taking your advice about the list of committors to the gitlab repository (plus Kenneth) to be the list of persons who needed to be asked for approval. But now there are these new concerns about whether someone was excluded by doing it like this, and there are also e.g. @bingao's comments about the proposal text itself in addition to the other comments, and in total I don't feel like I know anything about this clearly anymore. 

@bingao and @bast, concerning derivative work, my recollection about this is a little unclear, but I think that @ajthorva and @KennethRuud at some point discussed their intent to implement some kind of general-order code. I would also say that @ajthorva's code did have some generality to it, but in my opinion not to the same extent as my recursive scheme. The question of whether my work is derivative or not is impossible for me to answer: I cannot imagine that I would have made my code if Andreas' code did not already exist, and I must have looked at his code during development and I might for all I know included some of his code in mine - I don't recall and at the time one might say that I maybe had more of a notion of "working in Andreas' code" than something else.

All in all, I feel more confused after the discussions in this thread than before them. I now feel that I neither have the competence nor authority to make a decision here - both about what exactly to do for the release and for any of the other non-urgent questions about the licensing etc.. Maybe this is a time to involve Kenneth and have him decide?

**Comment by bast** [comment]
I was also convinced until this weekend that:
- that this is clearly derivative code (but I am neither a lawyer nor do I know how the new code was written in practice)
- that this is not a problem since all authors that the code derived from were involved

Trivial change: a typo or a rename, not easy to classify but I think we would all recognize trivial changes when we see them.

**Comment by bast** [comment]
In an attempt to unconfuse: why does it matter whether the present code is derivative or not? Has anybody been left out who has written code that the present code would have derived from if we consider it as derived? I don't think so. I don't think it matters.

**Comment by bast** [comment]
My feeling from this enormously long and also a bit tiring thread is that the issue is not the present and the past, I think all has been handled correctly, all persons have been involved and there is nothing to object in the licensing to LGPL but my feeling is that there are different views on authorship, credit, and in particular on authorship and credit in the future and this will probably require that Magnus, Gao, and Kenneth discuss live or over video, first.

**Comment by ringholm** [comment]
I don't know if or why it matter whether the present code is derivative or not. As to whether or not anyone has been left out: I mention some people earlier in this thread who may have contributed at least to the legacy code. The fact that I didn't manage to bring this up sooner and the various other concerns about this process makes me question my own ability to manage this aspect of the work and makes me worry that there are other things about this that I have overlooked.

When you say that Gao, Kenneth and I should discuss first, do you mean before this (LSDalton) release or, say, before the patch release?

In summary, it doesn't look like anyone here wants to make the decision to go ahead with the minting, DOI, release etc. Maybe I'll call Kenneth and ask him to decide.

**Comment by bast** [comment]
> When you say that Gao, Kenneth and I should discuss first, do you mean before this (LSDalton) release or, say, before the patch release?

... before we can conclude on the authorship and credit questions. I feel we are turning in circles here and a discussion among you three will hopefully unblock this.

Whether the authorship and credit needs to be agreed upon before the LSDalton release: I don't know. As to my possible authorship and my possible credit: I don't mind at all if this is concluded after the LSDalton release. Obviously I cannot make here choices for others.

**Comment by KennethRuud** [comment]
I see this is by now a very long thread, and if it is possible, could someone summarise the key questions that they would like my input on in a concise manner?

**Comment by ringholm** [comment]
I'll call you and tell what I can. But I might have lost the overview and if someone else feels they can summarize it, please do so here.

**Comment by bingao** [comment]
My points are as follows:

1. For the current release, we do not have the problem as regarding license, because most people involved in the development have agreed on. Thanks to Radovan and Magnus.
2. I see two things are vague to me, and may affect future changes as regarding at least the license. One is what is “OpenRSP project”, another is who are the copyright holder of the current form of the OpenRSP repository.
3. If we can not give a clear answer about what OpenRSP project is, then how can we define who are the authors of OpenRSP project.
4. Because I am not a layer, I do not know if all the authors involved are all the copyright holder of the current form of the OpenRSP repository. If yes, I have to disagree on the case if there is no derivative work left from persons in the current form of the OpenRSP repository.
5. Please note that I only focus on two things in point 4, one is the OpenRSP repository (not the whole project), second is its current form (source codes, documentation, and other files in the repository).
6. As regarding derivative work, I tried to read LPGL 2.1. My understanding is that it does not only mean left codes/documentation etc., it also involves other situations, such as Andreas’ and Magnus’ codes, in case that Magnus might include some lines from Andreas’ codes. But if one only learns design and algorithms from a code, and redesigns and rewrites, then it is not a derivative work. Please read LGPL for more information, and another design method to circumvent the copyright problem (https://en.m.wikipedia.org/wiki/Clean_room_design).
7. I chose Andreas’ and Magnus’ codes as example, because I thought it might be easier for us to distinguish what is a derivative work and what is not. It does not mean I want to exclude Andreas from the copyright holder :-)

**Comment by ringholm** [comment]
I spoke with Kenneth just now and he is willing to make a decision about what to do. He will post it on this thread. If it is fine with you that we proceed according to this decision then I am also fine with that.

**Comment by bingao** [comment]
OK. But please bring in my points and concerns, in particular who are the copyright holder of the current release. Thank you.

**Comment by ringholm** [comment]
OK, I will be back in a few hours to answer any questions that may have come up.

**Comment by ajthorva** [comment]
PS: (this is of *no* relevance for the "Release authors" issue) For security I just read through the initial commits (by Radovan). They only contain code written by him, myself and Gao. From then on people made their own commits. So I think we should conclude that the LGPL transition was approved by all the required people.

**Comment by KennethRuud** [comment]
@bingao Could you try clarify your point 4 for me, I am not sure which way you ask the question......

**Comment by bingao** [comment]
Hei Kenneth,

After reading LGPL, I think the most important (at least for me) is who are the **copyright holder** of the **current form of the OpenRSP repository**. Copyright holders will have the right to influence the future of the OpenRSP repository.

From https://github.com/openrsp/openrsp/blob/master/README.rst, I understand the **copyright holders are the OpenRSP authors** (http://www.openrsp.org/en/latest/authors.html). These two terms have the same meaning in the current context. Currently there are 7 persons in the list of authors.

What I can accept is: if we want to add a person into this list (I understand the person will become the copyright holder), and if it means the person becomes the author of the **current form of the OpenRSP repository**, there must be derivative work from the person in the **current form of the OpenRSP repository**. Otherwise, I can not see the reason that the person can be a **copyright holder** of the **current form of the OpenRSP repository**.

**Comment by bingao** [comment]
Hei Kenneth,

One thing I forgot to mention. It is actually not difficult to check whose derivative work left in the **current form of the OpenRSP repository**.

I just checked all the **current files** in the repository:

1. Most CMake files are written by the current 7 authors. Questionable files include
   1. `cmake/ConfigParentSettings.cmake`, which I modified based on other project (probably) by @bast 
   2. `cmake/cmake_uninstall.cmake.in` is from http://cmake.org/Wiki/CMake_FAQ#Can_I_do_.22make_uninstall.22_with_CMake.3F, if it is a problem, we can simply remove it because it is not necessary to have the functionality of uninstallation
2. Files in `doc` are all written by the current 7 authors except for
    1. some coding standards added by @bast , that he knows who are the copyright holder
    2. `doc/conf.py` created by sphinx-quickstart, @bast might know the copyright holder of this file?
    3. files about how sphinx works added by @bast 
3. Files in `include` and `web` are all written by myself.
4. Files in `tests` are all written by myself. The questionable files are matrix files in `tests/DMatHF`, which are computed and generated using Dalton program. @bast  please correct me if I am wrong, I do not think these data/matrix files is copyrightable by Dalton program.
5. Files in `src` are written by the current 7 authors, except for these `*.f90` files in `src/ao_dens`, @ajthorva and @ringholm should be the best person to say who are the copyright holders of these `*.f90` files.

**Comment by KennethRuud** [comment]
I am not sure I in the end have the authoritative answer, but going specifically into what Bin Gao asks in terms of copyright authors to the repository and code in the repository.

In terms of the authorship of the repository, I can agree with @bingao general definition, and I note that github actually recommends defining this when establishing the repository.  In terms of whether there are some differences compared to the current author list, I guess the comments are the following:
- If you find Coriolis effects in the code, then Orian Louant should be a co-author. This would be in the VPT2, which perhaps is a separate code?
- You may want to check for hyper-Raman/Raman Optical Activity, whether there are routines by Yann Cornaton in there.
- And if there is could for EFISHG-CID, then possible there is something from Marco Anelli.

I guess it is a fair question whether I have any ownership to the repository. I feel ownership to the project per se, but not necessarily to the code. I am ok if I am not an author of the repository, but I would perhaps be less comfortable if this implied not being an author for the DOI for the release. In the context of the release, I would also also think that both @foeroyingur and @simensr have made significant contributions that could in principle warrant association with the DOI. I note that by default, Zenodo would take all user names associated with the repository, but that you can edit this later as also @bast  pointed out.

Does this make sense, or is there some inconsistency in this proposal? And did this address the concerns that are unresolved in this thread?

**Comment by ringholm** [comment]
For the files in `src/ao_dens` on the `master` branch, not counting `CMakeLists.txt` and files in the `adapter` subdirectory, I have the following overview:

```
File					Written by

rsp_choose_rule.f90			I think I (MaR) wrote all or practically all of it
rsp_contribs.f90			At least MaR, BG, RB, AJT, likely also NHL, JMHO, some code I don't know the exact origin of and which I think is quite old
rsp_field_tuple.f90			At least MaR, BG, DF
rsp_general.f90				At least MaR, BG, DF, maybe also lines from AJT
rsp_indices_and_addressing.f90		At least MaR, BG, maybe DF
rsp_lof_caching_tmp.f90			At least MaR, BG, maybe also lines from AJT
rsp_pert_table.F90			I think only BG
rsp_perturbed_matrices.f90		At least MaR, BG, maybe DF
rsp_perturbed_sdf.f90			At least MaR, maybe DF, BG, RB, NHL, JMHO, in particular not sure where all the lines in rsp_xcave_setup_dmat_perts came from
rsp_property_caching.f90		At least MaR, maybe BG, DF
rsp_sdf_caching.f90			At least MaR, BG, maybe also lines from AJT
```
As indicated by `at least`, I cannot guarantee that lines by other authors are not present, but at least this is what I can say from quickly scanning through these files now.

I cannot return to this until tomorrow morning or possibly later tonight.

**Comment by bingao** [comment]
A quick answer to @ringholm files in the `adapter` subdirectory are all written by me.

**Comment by bingao** [comment]
Hei Kenneth,

Thank you.

As regarding the contributions from Orian Louant, Yann Cornaton and Marco Anelli, I think @ringholm can answer whether their work and derivative work are still in the current code. I am sorry that I personally want to separate already removed codes (and authors with contributions totally removed from the current codes) from the current ones. It is just my opinion, I accept decisions from the majority of OpenRSP authors.

Moreover, if I remember correctly, some of these codes were moved into an OpenRSP postprocessing tool library. If yes, authors of these codes will definitely be the copyright holders of the tool library.

Last thing I would like to make it clear, so that I/we do not misunderstand: OpenRSP authors, copyright holders, authors in the release associated with the DOI, are these three terms the same or not? I feel it would be better that we clarify it.

**Comment by bast** [comment]
> Last thing I would like to make it clear, so that I/we do not misunderstand: OpenRSP authors, copyright holders, authors in the release associated with the DOI, are these three terms the same or not? I feel it would be better that we clarify it.

In my understanding this is the same group of people.

**Comment by bingao** [comment]
I also interpret that, (1) OpenRSP authors, (2) copyright holders and (3) authors in the release associated with the DOI are the same group of people. Therefore, I suggest we accept this argument unless someone objects to it.

Another suggestion according to the findings of @ringholm , I suggest that if we can not figure out all persons with work/derivative work in `src/ao_dens/*.f90`, we keep it open and write something together with the list of authors like "This might be an incomplete list of authors, and we are in the process of preparing the full list."

How do you think?

**Comment by ringholm** [comment]
To @bingao , I do not believe that any of the files I listed above contain contributions from Orian, Yann or Marco. I assume that the list on https://github.com/openrsp/openrsp/issues/43#issuecomment-439694848 refers to the `master` branch. 

There is definitely code on other branches where work by these three (or maybe not Marco) is still present - see e.g. `src/vib_pv_contribs.f90` (Yann was involved with this) and `src/openrsp.F90` (Orian was involved) on the branch `pcm/develop`. Right now I just got a message from Marco where he says that for EFISHG-CID, he only ran calculations. I am pasting the relevant part here with his approval since he mentions some other properties. I don't recognize having worked with them but if they were associated with OpenRSP before I became involved with OpenRSP, then that work (or work by other people that I don't know about) can still be present in e.g. the `src/legacy` folder of this and other branches.

Marco's replies about the properties he implemented (sorry that it's mostly in Norwegian):

"Jeg har implementert den quadrupole- quadrupole, dipolemagn-elquadrupole, magnetic quadrupole- electric -dipole, eloctopole-eldipole...jeg må sjekke faktisk..litt usikker akkurat nå . Men omtrent alle high order contributions which goes behind the dipole approximation that wasn't there from before

Om du tenke på enkelt properties da tror blir electric octopole magnetic quadrupole...resten var klar fra før , jeg har bare laget tensor"

Hope this was helpful and I will check in tomorrow morning.

**Comment by bingao** [comment]
Hei @ringholm 

Thank you for taking care of this. I think it is another ambiguous point to me: I am always talking about the current release or current files of the `master` branch. But it seems that you and @bast are talking about the whole repository with other branches considered. I do not know which one is more appropriate.

I understand you have your arguments, and I can accept the decision from the majority of current OpenRSP authors. I anyway need to express my arguments so that we have a big picture to consider:

If I understand the current issue as we are a new team to begin to rewrite OpenRSP and to release its `master` branch. There is neither work nor derivative work from Orian, Yann or Marco as you said. Moreover, my understanding is that their contributions are more related to post processing, and will be moved into https://gitlab.com/mringholm/rsp_toolbox (they will be definitely be the copyright holder of this RSP toolbox library). So it is quite probably that their contributions will never be part of OpenRSP. As such, I doubt they can be the copyright holders of OpenRSP.

If you decide to add them as the authors/copyright holders of OpenRSP, may I ask how you guys feel if as I said, their contributions will never be part of OpenRSP? Will you in the future remove their names from the list of authors/copyright holders? If yes, what are the guidelines to remove an author/copyright holder then? If you decide once the author, always the author, then the list of authors will grow and grow. We will encounter the same problem as Dalton has right now. I do not think it is good for any new author as @ajthorva already said.

**Comment by KennethRuud** [comment]
Although things simplify if all roles are the same, it is not quite clear to me either that they are. A repository may contain multiple libraries that potentially even include different licenses. As such, it could be strange if a library developer who provides functionality to the main project gets to strongly influence the development of the repository itself. 

In a similar manner, if someone starts working on a new branch, building on the OpenRSP respository and main branch, but otherwise not committing back to the main branch, it is ok that they are owners to the repository, but strange if they would be authors of an official release that originates from the main branch.

The way I understand GitHub, they assign a particular influence on repository ownership to the first version of the repository, since you then in a sense get "everything" with branches etc.

**Comment by bast** [comment]
For licensing, DOI, authors, and copyright we should IMO only consider the current `master` branch and its history. We can move all other branches into a separate repository under https://github.com/openrsp if it helps. And for me it is crystal clear that Kenneth should be part of DOI and authors and copyright.

**Comment by bast** [comment]
All related projects and libraries and tools should IMO be collected under https://github.com/openrsp so that they are easier to discover and find for users but they can carry different licenses, different copyrights, different authors.

**Comment by ringholm** [comment]
I don't have a strong opinion about any particular set of criteria for being an author at this point and I will go along with whatever you decide. 

Concerning the legacy code, I did raise a point about this (see the comments after https://gitlab.com/openrsp/lsdalton/merge_requests/32#note_102701599 ) because I thought it might be cleaner to take it out, but I was reassured by Radovan who replied that he did not consider it a problem. You can read the rest of that discussion too for more information about the circumstances.

I consider it unlikely that the contributions that we definitely know that Yann, Orian (and possibly Marco) made will be taken back into use in what we have called the OpenRSP "core routines". I think it is likely that any lines by NHL and JMHO that there may be in the files I list in https://github.com/openrsp/openrsp/issues/43#issuecomment-439712147 are not in use right now. I have just made a PR for the removal of rsp_contribs.f90 as it does not appear to contain any code that is  used at the moment.

I think that for the files I list in https://github.com/openrsp/openrsp/issues/43#issuecomment-439712147 on `master` it is likely that there are no further authors than the ones I mention. But this whole discussion has made me generally hesitant to be certain about anything concerning this and so I cannot say this with full certainty.

**Comment by foeroyingur** [comment]
The lines by NHL and myself (and Arnfinn) related to PE are not in use. They are still (privately) accessible through the private Dalton repo and the locked OpenRSP repo on GitLab. In my opinion this is good enough and this code can be removed from the GitHub repo.

**Comment by bingao** [comment]
I would like to make it clear: I apologize if it is my arguments that makes anybody thinks @KennethRuud can not be the author (of the release). Actually, all my arguments are for:

1. The scheme on how newly proposed person can be an author or not.
2. Even it is not urgent, but it came to that point eventually--do we need a scheme that removes a person from the list of authors?

Furthermore, I hope you have not misunderstood that, I did not discuss all current 7 authors, because I have no doubt that they are for sure the authors. @KennethRuud is the leader of the whole OpenRSP project. I am not sure if there is any code/documentation left from Dan. Even not, I remember he worked with @ringholm together for the recursive routines. So I will change my word from code/documentation to "intellectual property work" when we need to discuss all authors. Also, I did not mention @simensr and @foeroyingur because I know what contributions they have made, so no doubt either.

When we started to discuss other contributors, I did not think they played the same role as Kenneth. Therefore, I emphasize their work and derivative work instead of supervision, organization, leading, etc..

**Comment by bingao** [comment]
My last few arguments for voting for authors per official release:

1. I recall @simensr and Dalton community have done in such a way, right?
2. We were suggested that we need to remove some stale branches, but if we are going to add contributors only involved in these branches, isn't it contradictory?

The last point is only if you would like to add all authors from all branches in the official release, and would like to use LGPL 2.1 for the whole OpenRSP repository, then we should be aware of that there was no license at the beginning of the repository. As such, some branches do not have a clear statement as regarding the license. One example is `pcm/develop` (https://github.com/openrsp/openrsp/tree/pcm/develop).

**Comment by bast** [comment]
> We were suggested that we need to remove some stale branches, but if we are going to add contributors only involved in these branches, isn't it contradictory?

I don't understand - the official release that we are trying to create here only concerns `master`. To remove all "distraction" I would like to remove stale branches and also recommend to move all other branches to another repository. The risk however is that they are then out of sight and out of mind. But they will not be lost. For the authorship of this release we should only consider the `master` branch. In my understanding person X cannot be author of version Y if their code is not part of version Y.

**EDIT**: "if their code is not part of version Y" -> "if they have not contributed to version Y"

**Comment by bingao** [comment]
Thank you @bast. Your points, I totally agree with.

**Comment by ringholm** [comment]
Just mentioning it to collect the information here that Karen made some of the OpenRSP tests on the LSDalton side.  She is now being added as an LSDalton author. Does this work by her imply that she should also be added as an OpenRSP author on this release?

**Comment by ringholm** [comment]
@foeroyingur Is it correct that no code by you or Nanna is presently on the `master` branch of this repository?

**Comment by bingao** [comment]
Accordingly to Radovan’s suggestion that I also agree with: I would like to say if her codes are on the master branch of OpenRSP, then yes.

But please note: it is now only Radovan and myself agree with the above decision. I would like other authors also to express their opinion.

**Comment by foeroyingur** [comment]
Yes, that is correct. I am here referring to PE related code and do not know whether Nanna has made contributions in other directions. We had some PE related code in the old interfaces subdirectory and then some calls to that code in various files but this is no longer present, to the best of my knowledge.

I made a very trivial fix in 709f0ac8d51316a421cf3d16b7102aeaf35b2fa5 but this does not warrant authorship or anything like that.

**Comment by ringholm** [comment]
I don't think that Nanna has done any non-PE/external electric field-related work in OpenRSP. I don't think Karen has done any work inside the OpenRSP repository. I asked her and she should hopefully reply to the list.

I will go out soon - if anyone feels that the time is ready to make a decision, please feel free to do so without me. I will be back this evening but if there are any pressing matters I can be reached on the phone at (+47) 48104790.

**Comment by ajthorva** [comment]
> But please note: it is now only Radovan and myself agree with the above decision. I would like other authors also to express their opinion.

(after Radovan's EDIT I'm not sure Gao agrees with him any longer)

I agree with Radovan after the EDIT. Besides the master (code, test code and documentation), I would also consider extensive testing, funding, organisation and leadership of OpenRSP as "contributions" worth authorship on the release's DOI.

How to manage release authors in the long run *you* will have to decide between this and the next release - but please not now, as this discussion is starting to hold back the release of LSDalton

**Comment by bingao** [comment]
I did not see the EDIT. But I still agree with it. In my last two posts, I have suggested that we use "intellectual property work" instead of codes/documentation. I guess it is the same meaning as "contributions".

I agree that we should not hold back the release of LSDalton due to this not easily solved problem. But I suggest that we should be cautious to add new authors. I would like to keep it as an open issue (if allowed), by writing something together with the list of authors: "This might be an incomplete list of authors, and we are in the process of preparing the full list." Because I feel that no one likes to the person to remove an author if he/she is not qualified any more.

**Comment by bast** [comment]
> we should be cautious to add new authors

For this release? Or for future releases? If for future, let us please deal with it later. Let us please conclude this. This DOI is about this release. There are hopefully many more releases to come. If we realize that somebody was left out (who?) we can fix that.

> no one likes to the person to remove an author if he/she is not qualified any more

Let us talk about this later. Are we worried that somebody is missing? Then we will fix that. Are we worried that the list is too wide?

I want to set the tag and DOI for reproducibility's sake. This will sadly not have any other effect than that. It will not increase our h-indices. Software citation is not important yet. Sadly. One day it will be but then we will be 15 releases further down the road.

If our worry is complication of a future license change:
- I don't think we need to change the license any time soon
- If we do, I don't think it would be difficult
- Let us worry about it later

**Comment by bingao** [comment]
> > we should be cautious to add new authors
> 
> For this release? Or for future releases? If for future, let us please deal with it later. Let us please conclude this. This DOI is about this release. There are hopefully many more releases to come. If we realize that somebody was left out (who?) we can fix that.

Of course I am talking about this release.

OK. Let me be the person to conclude the author issue for the current release because it seems @ringholm will be back until evening. I will take the responsibility and blame :-)

1. I understand @bast will move all other branches into another repository, and we are releasing based on the `master` branch.
2. New authors/copyright holders so far include @simensr and @foeroyingur

**Comment by foeroyingur** [comment]
Just to be clear. I should **not** be added as author/copyright holder since I haven't made any significant contributions.

**Comment by bingao** [comment]
Oops, added. Please do significant contributions from now on :-)

**Comment by bingao** [comment]
Hei @bast 

I just made a pull request to add @simensr and @foeroyingur as authors. If you agree, please confirm the request.

**Comment by bast** [comment]
The important question for me is whether also Simen and Jógvan Magnus agree (see #73). If yes, I will press the green button.

**Comment by ringholm** [comment]
Just checking in - I see that @bingao has concluded and I am happy for this taking of responsibility. It looks to me like @bast needs the approval of @simensr and @foeroyingur to proceed. I will e-mail them and ask them to look here.

I also notice that "extensive testing" is mentioned by @ajthorva as something which should lead to authorship. I would just remind, like I believe was told earlier, that Karen has been involved with creating tests for OpenRSP in LSDalton. She is being made an author of LSDalton for this but Andreas' statement may raise the question of whether she should also be an OpenRSP author. I am her co-supervisor and I will take myself out of the discussion of this matter. Furthermore, the people who have done calculations with OpenRSP (the recursive version in Dalton) have definitely (Yann) or may have (Marco and possibly other) done some work (e.g. numerical differentiation) to verify properties. I am not sure if this testing should qualify as extensive and it was at least partly with the goal of running production calculations - but some of the properties they calculated may not have been thoroughly tested before they did it. It may be that you conclude that this does not qualify for authorship but since this remark was raised by Andreas I feel obligated to mention it. I don't feel competent to decide this.

I will be back later this evening if there are further comments.

**Comment by bingao** [comment]
Hei @ringholm

You do not need to write to @simensr and @foeroyingur the issue is solved.

As regarding authorship criteria, I need a rest to rethink them. But not for the current release, probably for next release.

**Comment by ringholm** [comment]
Okay, then I assume all has been decided for this release? I am at dinner now but will be on standby when I get back until midnight.

**Comment by bingao** [comment]
Hei Magnus,

Decision made so far:

1. @simensr is added as new author/copyright holder

2. @bast will move all other branches into another repository, and we are releasing based on the master branch

**Comment by bast** [comment]
I am working on it. Running some tests. I will update everybody via the mailing list about the branch cleanup.

**Comment by ringholm** [comment]
OK, that sounds good. We get the opportunity to make changes if we need to for the patch release later. I also want to say thanks a lot for all of the excellent effort with this - I am truly grateful for it.

**Comment by simensr** [comment]
Let me join in. Thanks a lot for all the excellent work! These have been a stressful few months, but we made it very far and should be proud:-)

I have also launched several tests now, and if they run as expected the Dalton release will be official later this evening.

**Comment by bast** [comment]
Once Simen tags LSDalton, I will tag OpenRSP. The DOI can come 1 day later, does not matter.

**Comment by ringholm** [comment]
@bast just double-checking that this does not conflict with https://gitlab.com/dalton/lsdalton/issues/45

**Comment by bast** [comment]
After my last post here I remembered to enable Zenodo since it will create DOI automatically when I tag the code so we will have it in not too long.

**Comment by ringholm** [comment]
OK, great!

**Comment by ringholm** [comment]
Is this issue settled and ready to close now?

**Comment by bast** [comment]
It would be good if Zenodo and the DOI are mentioned here: http://www.openrsp.org/en/latest/citations.html - after that it can be closed.

**Comment by ringholm** [comment]
Decided today: Will try to settle during this week.

**Comment by ringholm** [comment]
settle before release

**Comment by ringholm** [comment]
@bast will do

**Comment by bast** [comment]
Are there any changes in the list of authors compared to this: https://openrsp.org/en/latest/authors.html

**Comment by ringholm** [comment]
No, these are all the authors as far as I'm aware. Please note that Daniel is to my knowledge not affiliated with a university anymore but I don't know his present affiliation.

**Comment by bast** [comment]
My understanding of the steps left here:
- Keep author list of https://openrsp.org/en/latest/authors.html
- Assign a DOI
- Reference the DOI on https://www.openrsp.org/en/latest/citations.html with a recommended citation for the software

**Comment by ringholm** [comment]
That's the same understanding that I have!
